src/timeseries.jl

```diff
@@ -0,0 +1,185 @@
+export veltimeseries
+
+"""
+    veltimeseries(notes::MIDI.Notes, grid = 0:1//notes.tpq:1; zeros::Bool = false, av::Bool = false)
+
+Generate a time series of velocities out of given notes.
+Arguments:\n
+- `notes`:  the notes from which the time series is generated
+- `grid`:   give a grid to which the notes will be quantized before time
+            series generation. See [`quantize`](@ref) for details.
+- `zeros`:  put zeros on gridpoints/ticks where no note occurs
+- `av`:     If multiple notes appear at one gridpoint, the highest velocity is
+            chosen by default. Set this to true to average over velocities at
+            that grid point
+"""
+function veltimeseries(notes::MIDI.Notes, grid = 0:1//notes.tpq:1; zeros::Bool = false, av::Bool = false)
+
+    #quantize if needed
+    if grid != 0:1//notes.tpq:1
+        quantize!(notes, grid)
```

Comment:
1. You should never operate in-place and not state it in the documentation. This is not performance critical, just use `quantize`.
2. You should always `quantize`. The user *can* actually change the `tpq` without changing the note positions, which means that they are not quantized to this "default" grid.